src/Kiosk.Api/Controllers/EctsSubjectController.cs

```diff
@@ -215,17 +216,18 @@ public async Task<IActionResult> GetSubjectsByMajor([FromQuery] EctsSubjectReque
     /// <response code="500">Internal Server Error</response>
     /// <response code="404">Not Found</response>
     /// <returns>The confirmation that the request was processed</returns>
-    [HttpGet("/years")]
+    [HttpGet("years")]
     [Consumes("application/json")]
     [Produces("application/json")]
     [ProducesResponseType(typeof(IEnumerable<int>), StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
     [ProducesResponseType(typeof(IEnumerable<int>), StatusCodes.Status500InternalServerError)]
-    public async Task<IActionResult> GetYears(BaseEctsSubjectRequest baseEctsSubjectRequest, CancellationToken cancellationToken)
+    public async Task<IActionResult> GetYears([FromQuery, Required] BaseEctsSubjectRequest baseEctsSubjectRequest, CancellationToken cancellationToken)
     {
         try
         {
-            var result = await _ectsSubjectRepository.GetYears(baseEctsSubjectRequest, cancellationToken);
+            var result = (await _ectsSubjectRepository.GetYears(baseEctsSubjectRequest, cancellationToken))
+                .OrderDescending();
```

Comment:
perhaps you could move this to service, this way `OrderDescending` will be in service layer